jupyterhub/templates/hub/deployment.yaml

```diff
@@ -46,6 +46,11 @@ spec:
                     - key: release
                       operator: In
                       values: [{{ .Release.Name | quote }}]
+      {{- $_ := merge (dict "podKind" "core") . }}
+      {{- $dummy := include "jupyterhub.prepareScope" $_ }}
+      {{- if $_.hasTolerations }}
+      {{- include "jupyterhub.tolerations" $_ | nindent 6 }}
+      {{- end }}
```

Comment:
prepareScope prepopulates $_ with various stuff, and then the `jupyterhub.tolerations` template utilizes that. The $_ will be reused by a future `jupyterhub.affinity` template so it makes sense to prepare the scope once instead of twice.